qutebrowser/mainwindow/prompt.py

```diff
@@ -630,6 +630,20 @@ def __init__(self, question, parent=None):
 
         self._to_complete = ''
 
+    def _directories_hide_show_model(self, path):
+        """Get rid of non-matching directories."""
+        try:
+            num_rows = self._file_model.rowCount(self._root_index)
+            for row in range(num_rows):
+                index = self._file_model.index(row, 0, self._root_index)
+                hidden = self._to_complete not in index.data()
+                if index.data() == '..':
+                    hidden = False
```

Comment:
You could simplify this a bit:

```suggestion
                filename = index.data()
                hidden = self._to_complete not in filename and filename != '..'
```